File: linked-list/operations/searching/searching.py
class Node:
    
    def __init__(self,data):
        self.data = data
        self.next = None

#main
for i in range(0,5):
    if i == 0:
        temp = Node(10)
        head = temp
        tail = temp
        
    else:
        temp = Node(i)
        tail.next = temp
        tail = temp

def searching(key,head):
    temp = head
    while temp!= None:
        if temp.data == key:
            return True
        temp = temp.next
    
    return False

if searching(11,head):
    print("element available")
else:
    print("not available")

File: linked-list/operations/searching/searching.c
#include<stdio.h>
#include<stdlib.h>

typedef struct node
{
    int data;
    struct node* next;
}node;

int searching(int key,node* head);

int main()
{
    node* head = malloc(sizeof(node));
    head->data = 10;
    node* tail = head;
    node* temp = NULL;
    for(int i= 0; i<5;i++)
    {
        temp = malloc(sizeof(node));
        tail->next  = temp;
        temp->data = i*10;
        tail = temp;
    }
    
    if(searching(10,head) == 0)
    {
        printf("Element Available");
    }
    else
    {
        printf("Not available");
    }
}

int searching(int key,node* head)
{
    node* p = head;
    
    while(p!=NULL)
    {
        if(p->data == key)
        {
            return 0;
        }
        
        p = p->next;
    }
    p = NULL;
    return 1;
}